TypeCobol/Compiler/Parser/ProgramClassParserStep.cs

```diff
@@ -85,21 +86,25 @@ public static void ParseProgramOrClass(TextSourceInfo textSourceInfo, ISearchabl
 
             //Create link between datas
             programClassBuilder.SyntaxTree.Root.AcceptASTVisitor(new TypeCobolLinker());
+            //Complete some information on Node and run checker that need a full AST
+            programClassBuilder.SyntaxTree.Root.AcceptASTVisitor(new Cobol85CompleteASTChecker());
+            programClassBuilder.SyntaxTree.Root.AcceptASTVisitor(new DiagnosticsChecker(diagnostics));
```

Comment:
As discussed today, you can move this DiagnosticsChecker to method AllDiagnotics of CompilationDocument.